code/modules/power/groundmap_geothermal.dm

```diff
@@ -357,7 +352,239 @@ GLOBAL_VAR_INIT(generators_on_ground, 0)
+	desc = "A marvel of modern engineering and a shining example of pioneering bluespace technology, able to power entire colonies with very little material consumption - perfectly suited for isolated areas on the outer rim.\nHighly volatile, but that shouldn't matter on some quiet backwater colony, right..?"
+	icon = 'icons/obj/machines/tbg.dmi'
+	power_generation_max = 5000000 //Powers an entire colony
+	time_to_break = 5 SECONDS
+	voice_filter = "alimiter=0.9,acompressor=threshold=0.2:ratio=20:attack=10:release=50:makeup=2,highpass=f=1000"
+	//Stores whether we're in the turning off animation
+	var/winding_down = FALSE
+	//List of turbines connected for visuals
+	var/list/connected_turbines = list()
+	//Ambient soundloop
+	var/datum/looping_sound/generator/tbg/ambient_soundloop
+	//Explosion alarm soundloop
+	var/datum/looping_sound/alarm_loop/generator/alarm_soundloop
+
+	COOLDOWN_DECLARE(toggle_power)
+
+/obj/machinery/power/geothermal/tbg/Initialize()
+	. = ..()
+	ambient_soundloop = new(list(src), is_on)
+	alarm_soundloop = new(list(src), buildstate == GENERATOR_EXPLODING)
+	for(var/direction in GLOB.cardinals)
+		var/obj/machinery/power/tbg_turbine/potential_turbine = locate(/obj/machinery/power/tbg_turbine, get_step(src, direction))
+		if(!potential_turbine)
+			continue
+		connected_turbines += potential_turbine
+		potential_turbine.connected = src
+
+/obj/machinery/power/geothermal/tbg/Destroy()
+	QDEL_NULL(ambient_soundloop)
+	QDEL_NULL(alarm_soundloop)
+	. = ..()
+
+/obj/machinery/power/geothermal/tbg/update_icon_state()
+	. = ..()
+	if(winding_down)
+		icon_state = "on25"
+	if(buildstate == GENERATOR_EXPLODING)
+		icon_state = "exploding"
+	for(var/obj/machinery/power/tbg_turbine/turbine in connected_turbines)
+		turbine.update_icon()
+
+/obj/machinery/power/geothermal/tbg/interact_hand(mob/living/user)
+	if(!COOLDOWN_CHECK(src, toggle_power))
+		balloon_alert(user, "Busy")
+		return
+	return ..()
+
+/obj/machinery/power/geothermal/tbg/turn_on()
+	COOLDOWN_START(src, toggle_power, 10 SECONDS)
+	ambient_soundloop.start()
+	SSmachines.active_bluespace_generators++
+	SEND_SIGNAL(SSmachines, COMSIG_GLOB_BLUESPACE_GEN_ACTIVATED)
+	return ..()
+
+/obj/machinery/power/geothermal/tbg/turn_off()
+	COOLDOWN_START(src, toggle_power, 10 SECONDS)
+	. = ..()
+	winding_down = TRUE
+	update_icon()
+	addtimer(CALLBACK(src, PROC_REF(finish_winding_down)), 10 SECONDS)
+	ambient_soundloop.stop()
+	SSmachines.active_bluespace_generators--
+	if(!SSmachines.active_bluespace_generators)
+		SEND_SIGNAL(SSmachines, COMSIG_GLOB_ALL_BLUESPACE_GEN_DEACTIVATED)
+
+/// Updates the turbine animation after the winding down sound effect has finished
+/obj/machinery/power/geothermal/tbg/proc/finish_winding_down()
+	winding_down = FALSE
+	update_icon()
+
+
+/obj/machinery/power/geothermal/tbg/damage_generator()
+	if(buildstate >= GENERATOR_EXPLODING) //Already exploding; can't be damaged more than that!
+		return FALSE
+	if(!..())
+		return FALSE
```

Comment:
prefer . = ..() usage

Reply:
its safer if theres a runtime and it means less boilerplate changes when the code is chaged later